### PR TITLE
ASM-9481 Windows 2008 R2 VM Deployment on VDS Cluster Fails with NilClass error

### PR DESCRIPTION
**Pull request by sushilrai**
VMware ESXi discovery is failing when there are one or more host in non-responding state. Added code logic to skip port-group information in case hosts are in non-responding state

### CODE REVIEW

File: bin/discovery.rb [hunk 1]
```diff
@@ -229,6 +229,9 @@ def collect_vds_portgroup_attributes(portgroup)
 end
 
 def collect_portgroup_attributes(network_obj)
+  # In case ESXi server is in non-responding state then need to skip port-group information
+  return {} unless network_obj.host[0].configManager
+
   network = network_obj.host[0].configManager.networkSystem.networkInfo.portgroup.select { |x| x.spec.name == network_obj.name }
```
Comment (gavin-scott) [inline bin/discovery.rb]: In Sumitha's case there was a `network_obj.host[1]` that did have a non-null `configManager`. How do we know which host to look at?

Comment (gavin-scott) [inline bin/discovery.rb]: Put another way, why should we always look at `host[0]`?

Comment (sushilrai) [inline bin/discovery.rb]: We are calling method `collect_portgroup_attributes` for each host in the cluster. Here we pass the network object for the host for which inventory is getting collected.

File: bin/discovery.rb [hunk 2]
```diff
@@ -228,8 +228,13 @@ def collect_vds_portgroup_attributes(portgroup)
...
 
-def collect_portgroup_attributes(network_obj)
-  network = network_obj.host[0].configManager.networkSystem.networkInfo.portgroup.select { |x| x.spec.name == network_obj.name }
+def collect_portgroup_attributes(network_obj, parent)
+  # In case ESXi server is in non-responding state then need to skip port-group information
+  network_host = network_obj.host.select {|h| h.name == parent.name}.first
+  return {} if network_host.summary.runtime.connectionState == "disconnected"
+  return {} unless network_host.configManager
+
+  network = network_host.configManager.networkSystem.networkInfo.portgroup.select { |x| x.spec.name == network_obj.name }
```
Comment (gavin-scott) [inline bin/discovery.rb]: Nice, thanks for updating this.

**Pull request merged**